File: app/src/main/java/foocafe/org/foocafe/SignUpActivity.java
package foocafe.org.foocafe;

import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Toast;

public class SignUpActivity extends AppCompatActivity {

    private WebView web;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_up);
        final String eventUrl = getIntent().getStringExtra("url");
        Log.i("WEBB", eventUrl);

        web = (WebView) findViewById(R.id.webView);
        web.getSettings().setJavaScriptEnabled(true);
        web.loadUrl(eventUrl + "/registrations/new");
        web.setWebViewClient(new WebViewClient() {

            @Override
            public void onPageFinished(WebView view, String url) {
                if (!url.equals(eventUrl + "/registrations/new")) {
                    Toast.makeText(SignUpActivity.this, "Redirecting", Toast.LENGTH_SHORT).show();
                    new Handler().postDelayed(new Runnable() {
                        @Override
                        public void run() {
                            finish();
                        }
                    }, 2500);
                }
            }

        });
    }
}

File: app/src/main/java/foocafe/org/foocafe/SignUpActivity.kt
package foocafe.org.foocafe

import android.annotation.SuppressLint
import android.os.Handler
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.Toast

class SignUpActivity : AppCompatActivity() {

    @SuppressLint("SetJavaScriptEnabled")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sign_up)
        val eventUrl = intent.getStringExtra("url")

        val web = findViewById<WebView>(R.id.webView)
        web.settings.javaScriptEnabled = true
        web.loadUrl(eventUrl + "/registrations/new")
        web.webViewClient = object : WebViewClient() {

            override fun onPageFinished(view: WebView, url: String) {
                if (url != eventUrl + "/registrations/new") {
                    Toast.makeText(this@SignUpActivity, "Redirecting", Toast.LENGTH_SHORT).show()
                    Handler().postDelayed({ finish() }, 2500)
                }
            }
        }
    }
}